### ISSUE TITLE
Unnecessary header reset causes direction indicator to revert

### ISSUE THREAD

**Issue by DanStout** [ISchwarz23/SortableTableView#68]
I'm setting up some data on a fragment which isn't initially visible, and calling `sort(0, false)` on a `SortableTableView` with the goal of having the data sorted and the little triangle indicator correctly showing the default sort (SORTED_DESC) once the fragment becomes visible. In actuality, however, the column sort indicator is in the SORTABLE state (neither triangle is lit) once the fragment becomes visible.

For context, `TransactionsTableAdapter` extends `LongPressAwareTableDataAdapter` and `TransactionsTableView` extends `SortableTableView`, but they are pretty simple and I don't think they're the issue. Here is where the fragment sets up the adapter:

```
List<Transaction> transactions = //... from async operation
TransactionsTableAdapter adap = new TransactionsTableAdapter(getActivity(), transactions, mTableView);
mTableView.setDataAdapter(adap);
mTableView.sort(0, false);
```

I've been doing some poking around, and I think the issue is as follows:

- My call to `sort(int, boolean)` works correctly, and after the sorting, the `SortableTableHeaderView#setSortState(int, SortState)` method calls `ImageView#setImageResource(int)` for column 0 with the correct resource: `R.mipmap.ic_light_sorted_desc`
- Everything would be fine if the header stayed as it was. However, the following code in the constructor of `TableHeaderView` causes the headers to all be reset back to the "Sortable" state:

```
addOnLayoutChangeListener(new OnLayoutChangeListener() {
	@Override
	public void onLayoutChange(View v, int left, int top, int right, int bottom, int oldLeft, int oldTop, int oldRight, int oldBottom) {
		if (right - left != oldRight - oldLeft) {
			System.out.println("was resized");
			renderHeaderViews();
		}
	}
});
```

- At some point (maybe due to how the fragment isn't visible?) the listener is called, where the old layout values are all 0 and the new values are the actual dimensions, which means "was resized" is printed, `SortableTableHeaderView#renderHeaderViews()` is called, and the `SortState` of the columns are reset to `SORTABLE` - even column 0, which should be `SORTED_DESC`.

I'm not sure of the best way to fix this, but I think the sort states of the columns shouldn't be reset just because the header dimensions changed.

**Comment by ISchwarz23** [comment]
Hi @DanStout,

thanks to your deep investifation I was able to reproduce it very easy and in deed, it is a bug in the new version of the SortableTableView. I'm going to provide a bug fix for this very soon.

Best regards,
Ingo

**Comment by ISchwarz23** [comment]
Hi @DanStout,

this bug is now fixed in version 2.4.2

Best regards,
Ingo
